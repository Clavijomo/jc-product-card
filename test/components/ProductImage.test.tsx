import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/Products';

describe('ProductImage', () => {
    test('Debe mostrar el componente correctamente con la imagen personalizada', () => {
        const wrapper = renderer.create(
            <ProductImage img={product2.img} className='custom-image' />
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Mostrar el componente con la imagen del producto', () => {
        const wrapperImage = renderer.create(
            <ProductCard product={product2}>
                {() => (
                    <ProductImage />
                )}
            </ProductCard>
        );

        expect(wrapperImage.toJSON()).toMatchSnapshot();
    })
});