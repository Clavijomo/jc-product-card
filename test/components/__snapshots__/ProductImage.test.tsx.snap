// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="product"
  className="productImg custom-image"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage Mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
